File: frontend/src/types/image.ts
export interface ImageMedia {
  id: string;
  project_id: string;
  user_id: string;
  filename: string;
  original_filename?: string;
  file_size?: number;
  width?: number;
  height?: number;
  format?: string;
  s3_url?: string;
  thumbnail_url?: string;
  created_at: string;
}

export interface ImageAdjustments {
  brightness: number;
  contrast: number;
  saturation: number;
  temperature: number;
  exposure?: number;
  filter?: string;
  depthDisplacement?: number;
  parallaxTilt?: number;
  hologramScan?: boolean | number;
  particleDensity?: number;
  particleType?: string;
  volumetricLighting?: boolean | number;
  lightColor1?: string;
  lightColor2?: string;
  selected3DObject?: string;
}

File: backend/app/schemas/image.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from datetime import datetime

class ImageBase(BaseModel):
    filename: str
    original_filename: Optional[str] = None
    format: Optional[str] = None

class ImageCreate(ImageBase):
    project_id: str
    file_size: Optional[int] = None
    width: Optional[int] = None
    height: Optional[int] = None

class ImageEditRequest(BaseModel):
    edit_type: str  # enhance, filter, remove_background, adjust, crop
    prompt: Optional[str] = None
    parameters: Optional[Dict[str, Any]] = None

class ImageResponse(ImageBase):
    id: str
    project_id: str
    user_id: str
    file_size: Optional[int] = None
    width: Optional[int] = None
    height: Optional[int] = None
    s3_url: Optional[str] = None
    thumbnail_url: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True
